tests/integration/common-operator-integ-suite.sh

```diff
@@ -14,101 +14,146 @@
+  # Build and push docker image
+  # Notes: to be able to build and push to the local registry we need to set these variables to be used in the Makefile
+  # IMAGE ?= ${HUB}/${IMAGE_BASE}:${TAG}, so we need to pass hub, image_base, and tag to be able to build and push the image
+  IMAGE=${HUB}/${IMAGE_BASE}:${TAG} make docker-build docker-push
+}
+
+deploy_operator() {
+  echo "Deploying Operator"
+  make -s --no-print-directory deploy
+}
 
-echo "Deploy Istio"
-$COMMAND get ns "${CONTROL_PLANE_NS}" >/dev/null 2>&1 || $COMMAND create namespace "${CONTROL_PLANE_NS}"
-$COMMAND apply -f "${ISTIO_MANIFEST}" -n "${CONTROL_PLANE_NS}"
+check_ready() {
+  local NS=$1
+  local POD_NAME=$2
+  local DEPLOYMENT_NAME=$3
 
+  echo "Check POD: NAME SPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
+  timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
+    "until $COMMAND get pod --field-selector=status.phase=Running -n ${NS} | grep ${POD_NAME}; do sleep 5; done"
 
-echo "Check that Istio is running"
-check_ready "${CONTROL_PLANE_NS}" "istiod" "istiod"
+  echo "Check Deployment Available: NAME SPACE: \"${NS}\"   DEPLOYMENT NAME: \"${DEPLOYMENT_NAME}\""
+  $COMMAND wait deployment "${DEPLOYMENT_NAME}" -n "${NS}" --for condition=Available=True --timeout=${TIMEOUT}
+}
 
-echo "Make sure only istiod got deployed and nothing else"
-res=$($COMMAND  -n "${CONTROL_PLANE_NS}" get deploy -o json | jq -j '.items | length')
-if [ "${res}" != "1" ]; then
-  echo "Expected just istiod deployment, got:"
-  $COMMAND  -n "${CONTROL_PLANE_NS}" get deploy
-  exit 1
-fi
+main_test() {
+  # Add here all the validation tests for the operator
+  echo "Check that istio operator is running"
+  check_ready "${NAMESPACE}" "${DEPLOYMENT_NAME}" "${DEPLOYMENT_NAME}"
+
+  echo "Deploy Istio"
+  $COMMAND get ns "${CONTROL_PLANE_NS}" >/dev/null 2>&1 || $COMMAND create namespace "${CONTROL_PLANE_NS}"
+  $COMMAND apply -f "${ISTIO_MANIFEST}" -n "${CONTROL_PLANE_NS}"
+
+  echo "Check that Istio is running"
+  check_ready "${CONTROL_PLANE_NS}" "istiod" "istiod"
+
+  echo "Make sure only istiod got deployed and nothing else"
+  res=$($COMMAND -n "${CONTROL_PLANE_NS}" get deploy -o json | jq -j '.items | length')
+  if [ "${res}" != "1" ]; then
+    echo "Expected just istiod deployment, got:"
+    $COMMAND -n "${CONTROL_PLANE_NS}" get deploy
+    exit 1
+  fi
+
+  echo "Check that CNI daemonset is ready and running"
+  timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
+    "until $COMMAND rollout status ds/istio-cni-node -n ${NAMESPACE}; do sleep 5; done"
+}
 
-echo "Check that CNI deamonset ready are running"
-timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
-    "until $COMMAND  rollout status ds/istio-cni-node -n ${NAMESPACE}; do sleep 5; done"
+# SETUP
+check_arguments "$@"
+parse_flags "$@"
+initialize_variables
+get_internal_registry
+log_internal_registry
+build_and_push_image
+deploy_operator
+
```

Comment:
For operator test reuse with downstream pipeline, consider providing the ability to only run the tests. The operator deploy via subscription could be done within this suite or done so prior to running the test.

Reply:
Ok, that sounds reasonable to me. I'll add the capability to skip setup